### PR TITLE
darktile: init at 0.0.10

### PR DESCRIPTION
**Pull request by flexagoon**
###### Motivation for this change
Darktile is a GPU rendered terminal emulator designed for tiling window managers.
https://github.com/liamg/darktile

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  '';
+  dontInstall = true;
+
+  meta = with lib; {
+    description = "A GPU rendered terminal emulator designed for tiling window managers";
+    homepage = "https://github.com/liamg/darktile";
+    downloadPage = "https://github.com/liamg/darktile/releases";
+    changelog = "https://github.com/liamg/darktile/releases/tag/v${version}";
+    license = license.mit;
+    maintainers = with maintainers; [ flexagoon ];
```
Comment (bobby285271) [inline pkgs/applications/terminal-emulators/darktile/default.nix:65]: You should add yourself to [`maintainers/maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) with a separate commit `maintainers: add flexagoon`.

Comment (flexagoon) [reply pkgs/applications/terminal-emulators/darktile/default.nix:65]: I already did that in my other pull request
https://github.com/NixOS/nixpkgs/pull/135948

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    libXcursor
+    libXrandr
+    libXinerama
+    libXi
+    libXext
+    libXxf86vm
+    libGL
+  ];
+in
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:32]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  version = "0.0.10";
+
+  src = fetchFromGitHub {
+    owner = "liamg";
+    repo = "darktile";
+    rev = "v${version}";
+    sha256 = "0pdj4yv3qrq56gb67p85ara3g8qrzw5ha787bl2ls4vcx85q7303";
+  };
+
+  buildInputs = deps;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:43]:
```suggestion
  nativeBuildInputs = [ go pkg-config ];

  buildInputs = [
    libX11
    libXcursor
    libXrandr
    libXinerama
    libXi
    libXext
    libXxf86vm
    libGL
  ];
```

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  preBuild = ''
+    substituteInPlace scripts/build.sh \
+      --replace "bash" "sh"
+    makeFlagsArray+=(HOME=$TMP)
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:49]:
```suggestion
  postPatch = ''
    substituteInPlace scripts/build.sh \
      --replace "bash" "sh"
  '';

  makeFlags = [ "HOME=$TMP" ];
```

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  # Darktile does not have an 'install' Makefile target,
+  # so the installation is done in postBuild.
+  postBuild = ''
+    mkdir -p $out/bin
+    mv darktile $out/bin
+  '';
+  dontInstall = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:57]:
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/bin
    mv darktile $out/bin

    runHook postInstall
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by flexagoon** [comment]
I completely broke my git commit history in the process, but everything should be fixed @SuperSandro2000

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    export GOPATH=$TMP/go
+  '';
+
+  makeFlags = [ "HOME=$TMP" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv darktile $out/bin
```
Comment (sikmir) [inline pkgs/applications/terminal-emulators/darktile/default.nix:57]:
```suggestion
    install -Dm755 darktile -t $out/bin
```

**Comment by flexagoon** [comment]
@SuperSandro2000 Could you please review this?

**Comment by figsoda** [comment]
darwin build is failing, maybe you need `darwin.apple_sdk.frameworks.Carbon` in `buildInputs`

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 7]
```diff
@@ -0,0 +1,69 @@
+{ stdenv
+, buildGoModule
+, fetchFromGitHub
+, lib
+
+, go
+, pkg-config
+
+, libX11
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:9]:
```suggestion
, lib
, go
, pkg-config
, libX11
```

File: pkgs/applications/terminal-emulators/darktile/default.nix [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+, libXcursor
+, libXrandr
+, libXinerama
+, libXi
+, libXext
+, libXxf86vm
+, libGL }:
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/darktile/default.nix:16]:
```suggestion
, libGL
}:
```

**Pull request merged**